packages/mui-material/src/FilledInput/FilledInput.js

```diff
@@ -68,7 +68,7 @@ const FilledInputRoot = styled(InputBaseRoot, {
     },
     ...(!ownerState.disableUnderline && {
       '&:after': {
-        borderBottom: `2px solid ${theme.palette[ownerState.color].main}`,
+        borderBottom: `2px solid ${theme.palette[ownerState.color]?.main || 'primary'}`,
```

Comment:
```suggestion
        borderBottom: `2px solid ${theme.palette[ownerState.color || 'primary']?.main}`,
```

Reply:
Otherwise, it will end up as `2px solid primary`.

Reply:
Why set a default value in styles at all? In my opinion, it would be correct to simply set the default value in `props` and pass it to the `ownerState` object.

Reply:
We don't have a designs of how the `FilledInput` should look like if the color is `undefined` (this probably goes for other components too). How about we do: 
```suggestion
        borderBottom: `2px solid ${theme.palette[ownerState.color]?.main || theme.palette.divider}`,
```

Reply:
> @KlyshnikovRoman  Why set a default value in styles at all? In my opinion it would be correct to just set the default value color = 'primary' to props and pass it to the ownerState object.

I didn't take that approach because (1) it requires changes in `InputBase`, and (2) it creates inconsistency as `Input` and `OutlinedInput` both don't set the default value for `color` by themselves but rather done in `InputBase`.

Reply:
> @mnajdova We don't have a designs of how the FilledInput should look like if the color is undefined (this probably goes for other components too). How about we do:

We could, but for `Input` and `OutlinedInput`, if `undefined` is passed to `color`, it's default to primary color.

Reply:
I will follow Jun's suggestion and get the change merged if that's okay?